src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadmePreviewViewModel.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class ReadmePreviewViewModel : ViewModelBase
+    {
+        private INuGetPackageFileService _packageFileService;
+
+        public ReadmePreviewViewModel(INuGetPackageFileService packageFileService)
+        {
+            _packageFileService = packageFileService ?? throw new ArgumentNullException(nameof(packageFileService));
+            _errorLoadingReadme = false;
+            _canDetermineReadmeDefined = true;
+            _rawReadme = string.Empty;
+        }
+
+        private bool _errorLoadingReadme;
+
+        public bool ErrorLoadingReadme
+        {
+            get => _errorLoadingReadme;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _errorLoadingReadme, value);
+            }
+        }
+
+        private string _rawReadme;
+
+        public string ReadmeMarkdown
+        {
+            get => _rawReadme;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _rawReadme, value);
+            }
+        }
+
+        private bool _canDetermineReadmeDefined;
+
+        public bool CanDetermineReadmeDefined
+        {
+            get => _canDetermineReadmeDefined;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _canDetermineReadmeDefined, value);
+            }
+        }
+
+        public async Task LoadReadmeAsync(string rawReadmeUrl, CancellationToken cancellationToken)
+        {
+            var newReadmeValue = string.Empty;
+            var isErrorWithReadme = false;
+            bool canDetermineReadmeDefined = false;
+
+            if (!string.IsNullOrWhiteSpace(rawReadmeUrl))
+            {
```

Comment:
To reduce nested ifs we can take basic input validation checks like this and invert it to if (string.IsNullOrWhiteSpace(rawReadmeUrl)), set some defaults, and return.